### PR TITLE
Add a hypermodern cli-python 2.0.0

### PR DESCRIPTION
**Pull request by Zethson**
- [x] 2.0.0 template version bump
- [x] run pre-commit in the create_cli_python workflow
- [x] figure out a way to maybe not annoy all already existing projects
- [x] Run time name should be with hyphen and not with underscore

### CODE REVIEW

File: cookietemple/create/template_creator.py [hunk 1]
```diff
@@ -361,8 +361,8 @@ def create_dot_cookietemple(self, template_version: str):
         :param template_version: Version of the specific template
         """
         log.debug('Creating .cookietemple.yml file.')
-        self.creator_ctx.template_version = f'{template_version} # <<COOKIETEMPLE_NO_BUMP>>'
-        self.creator_ctx.cookietemple_version = f'{cookietemple.__version__} # <<COOKIETEMPLE_NO_BUMP>>'
+        self.creator_ctx.template_version = f"{template_version} # <<COOKIETEMPLE_NO_BUMP>>"
+        self.creator_ctx.cookietemple_version = f"{cookietemple.__version__} # <<COOKIETEMPLE_NO_BUMP>>"
```
Comment (Imipenem) [inline cookietemple/create/template_creator.py:365]: why `"` instead of `'`?

Comment (Zethson) [reply cookietemple/create/template_creator.py:365]: That was to satisfy prettier. We are now ignoring .cookietemple.yml -> I will revert this. Good catch.

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.bandit.yml [hunk 2]
```diff
@@ -1,7 +1,5 @@
-### profile may optionally select or skip tests
-
 # (optional) list included tests here:
 tests: []
 
 # (optional) list skipped tests here:
-skips: ['B403', 'B404', 'B603', 'B607']
+skips: ["B403", "B404", "B603", "B607"]
```
Comment (Imipenem) [inline cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.bandit.yml:5]: also here why the replacement?

Comment (Zethson) [reply cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.bandit.yml:5]: satisfy black for cli-python

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.flake8 [hunk 3]
```diff
@@ -0,0 +1,7 @@
+[flake8]
+select = B,B9,C,D,DAR,E,F,N,RST,S,W
+ignore = E203,E501,RST201,RST203,RST301,W503,D100
+max-line-length = 120
```
Comment (Imipenem) [inline cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.flake8:4]: think, with cookietemple we were at 160, you want to reduce it?

Comment (Zethson) [reply cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.flake8:4]: Yes. I personally like 160, but so many people think that it is too much. They're stuck with small laptops and want to code with 2 windows open at the same time.

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/dependabot.yml [hunk 4]
```diff
@@ -1,12 +1,45 @@
...
+      schedule:
+          interval: daily
+      open-pull-requests-limit: 10
+      target-branch: development
+      labels:
+          - DEPENDABOT
+      commit-message:
+          prefix: "[DEPENDABOT]"
+
+    - package-ecosystem: pip
```
Comment (Imipenem) [inline cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/dependabot.yml:14]: out of curiosity, why is this needed, like pip in GA?

Comment (Zethson) [reply cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/dependabot.yml:14]: pip=poetry in dependabots terms. It just needs to know what to look for

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/build_package.yml [hunk 5]
```diff
@@ -1,33 +1,41 @@
-name: Build {{ cookiecutter.project_slug }} Package
+name: Build {{ cookiecutter.project_name }} Package
```
Comment (Imipenem) [inline cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/build_package.yml:1]: Didn't we got rid of the project_name?

Comment (Zethson) [reply cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/build_package.yml:1]: Apparently not, because it would break elsewise :)

project_name allows spaces

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/constraints.txt [hunk 6]
```diff
@@ -0,0 +1,5 @@
+pip==20.3.3
```
Comment (Imipenem) [inline cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/constraints.txt:1]: we should add this to cookietemple dependabot config as well

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/publish_package.yml [hunk 7]
```diff
@@ -1,33 +1,50 @@
 name: Publish {{ cookiecutter.project_name }} to PyPI
 
 on:
-  release:
-    types: [created]
+    release:
+        types: [created]
```
Comment (Imipenem) [inline cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/publish_package.yml:5]: is this needed?

Comment (Zethson) [reply cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/publish_package.yml:5]: yes

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/constraints.txt (hunk 6 above)
Comment (Zethson) [reply cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/constraints.txt:1]: Those constraints are used for the GH workflows, not dependabot.

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/publish_package.yml [hunk 8]
```diff
@@ -1,33 +1,50 @@
...
-        # TODO COOKIETEMPLE: Configure your secrets to enable automatic deployment to PyPi on releases
-        # https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets
-        TWINE_PASSWORD: {% raw %}${{ secrets.PYPI_PASSWORD }}{% endraw %}
-      run: |
-        python setup.py sdist bdist_wheel
-        twine upload dist/*
+            - name: Install Poetry
+              run: |
+                  pip install --constraint=.github/workflows/constraints.txt poetry
+                  poetry --version
```
Comment (Imipenem) [inline cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/publish_package.yml:30]: this is the same as pip above, could this be somehow combined?

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/run_tests.yml [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+                          os: ubuntu-latest,
+                          session: "mypy",
+                      }
+                    - {
+                          python-version: 3.7,
+                          os: ubuntu-latest,
+                          session: "mypy",
+                      }
+                    - {
+                          python-version: 3.6,
```
Comment (Imipenem) [inline cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/run_tests.yml:41]: we may need to delete this here, as discussed earlier (drop python 3.6 support)

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/run_tests.yml [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+                          session: "tests",
+                      }
+                    - {
+                          python-version: 3.7,
+                          os: ubuntu-latest,
+                          session: "tests",
+                      }
+                    - {
+                          python-version: 3.6,
+                          os: ubuntu-latest,
```
Comment (Imipenem) [inline cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/run_tests.yml:62]: same here

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/run_tests.yml [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+
+            - name: Install Poetry
+              run: |
+                  pip install --constraint=.github/workflows/constraints.txt poetry
+                  poetry --version
+
+            - name: Install Nox
+              run: |
+                  pip install --constraint=.github/workflows/constraints.txt nox nox-poetry
+                  nox --version
```
Comment (Imipenem) [inline cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/run_tests.yml:116]: can the last three actions somehow be combined?

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/run_tests.yml [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
```
Comment (Imipenem) [inline cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/run_tests.yml:185]: same question as above?

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.pre-commit-config.yaml [hunk 13]
```diff
@@ -0,0 +1,51 @@
+repos:
+    - repo: local
+      hooks:
+          - id: black
```
Comment (Imipenem) [inline cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.pre-commit-config.yaml:4]: remove or not?

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/README.rst [hunk 14]
```diff
@@ -1,47 +1,75 @@
...
+.. |Python Version| image:: https://img.shields.io/pypi/pyversions/{{cookiecutter.project_slug }}
+   :target: https://pypi.org/project/{{ cookiecutter.project_slug }}
+   :alt: Python Version
+.. |License| image:: https://img.shields.io/github/license/{{ cookiecutter.github_username }}/{{ cookiecutter.project_slug }}
+   :target: https://opensource.org/licenses/{{ cookiecutter.license }}
+   :alt: License
+.. |Read the Docs| image:: https://img.shields.io/readthedocs/{{ cookiecutter.project_slug }}/latest.svg?label=Read%20the%20Docs
+   :target: https://{{cookiecutter.project_name}}.readthedocs.io/
+   :alt: Read the documentation at https://{{ cookiecutter.project_slug }}.readthedocs.io/
+.. |Build| image:: https://github.com/{{ cookiecutter.github_username }}/{{ cookiecutter.project_slug }}/workflows/Build%20{{ cookiecutter.project_slug }}%20Package/badge.svg
```
Comment (Imipenem) [inline cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/README.rst:18]: we should recheck, if `project_slug` is correct here

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.pre-commit-config.yaml (hunk 13 above)
Comment (Zethson) [reply cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.pre-commit-config.yaml:4]: I'd keep it for now. Just serves our idea.

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/run_tests.yml (hunk 9 above)
Comment (Zethson) [reply cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/run_tests.yml:41]: 3.10 will come out in April 

Ya, whatever we can drop 3.6

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/README.rst [hunk 15]
```diff
@@ -1,47 +1,75 @@
...
+   :target: https://pypi.org/project/{{ cookiecutter.project_slug }}/
+   :alt: PyPI
+.. |Python Version| image:: https://img.shields.io/pypi/pyversions/{{cookiecutter.project_slug }}
+   :target: https://pypi.org/project/{{ cookiecutter.project_slug }}
+   :alt: Python Version
+.. |License| image:: https://img.shields.io/github/license/{{ cookiecutter.github_username }}/{{ cookiecutter.project_slug }}
+   :target: https://opensource.org/licenses/{{ cookiecutter.license }}
+   :alt: License
+.. |Read the Docs| image:: https://img.shields.io/readthedocs/{{ cookiecutter.project_slug }}/latest.svg?label=Read%20the%20Docs
+   :target: https://{{cookiecutter.project_name}}.readthedocs.io/
```
Comment (Imipenem) [inline cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/README.rst:16]: why the `slug` in line 15 and `name` in 16?

Comment (Zethson) [reply cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/README.rst:16]: Mistake. Will fix.

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/cookietemple.cfg [hunk 16]
```diff
@@ -2,17 +2,17 @@
...
 [bumpversion_files_blacklisted]
+poetry = pyproject.toml
 
 [sync_level]
 ct_sync_level = minor
 
 [sync_files_blacklisted]
 changelog = CHANGELOG.rst
-requirements = requirements.txt
-requirements_dev = requirements_dev.txt
```
Comment (Imipenem) [inline cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/cookietemple.cfg:18]: we should now also adjust the linter!

Comment (Imipenem) [reply cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/cookietemple.cfg:18]: And sync also

Comment (Zethson) [reply cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/cookietemple.cfg:18]: I did mate, I did :)

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/noxfile.py [hunk 17]
```diff
@@ -0,0 +1,186 @@
...
+from pathlib import Path
+from textwrap import dedent
+
+import nox
+import nox_poetry.patch
+from nox.sessions import Session
+
+
+package = "{{ cookiecutter.project_slug_no_hyphen }}"
+python_versions = ["3.9", "3.8", "3.7", "3.6"]
```
Comment (Imipenem) [inline cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/noxfile.py:13]: drop 3.6 here as well

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/pyproject.toml [hunk 18]
```diff
@@ -0,0 +1,71 @@
...
+homepage = "https://github.com/{{ cookiecutter.github_username }}/{{ cookiecutter.project_slug }}"
+repository = "https://github.com/{{ cookiecutter.github_username }}/{{ cookiecutter.project_slug }}"
+documentation = "https://{{ cookiecutter.project_slug }}.readthedocs.io"
+{% if cookiecutter.project_slug != cookiecutter.project_slug_no_hyphen -%}
+packages = [
+    { include = "{{ cookiecutter.project_slug_no_hyphen }}", from = "src" },
+]
+{% endif -%}
+classifiers = [
+    "Programming Language :: Python :: 3.6",
```
Comment (Imipenem) [inline cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/pyproject.toml:17]: and drop it here

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/pyproject.toml [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+]
+
+[tool.poetry.urls]
+Changelog = "https://github.com/{{ cookiecutter.github_username }}/{{ cookiecutter.project_slug }}/releases"
+
+[tool.poetry.dependencies]
+python = "^3.6.1"
```
Comment (Imipenem) [inline cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/pyproject.toml:27]: and here

File: docs/available_templates/cli_python.rst [hunk 20]
```diff
@@ -10,89 +10,100 @@ but it may also be easily used as standard Python package without any command li
...
+1. `poetry <https://setuptools.readthedocs.io/en/latest/>`_ for code packaging
+2. `click <https://click.palletsprojects.com/>`_ or no command line interface
 3. `pytest <https://docs.pytest.org/en/latest/>`_ or `unittest <https://docs.python.org/3/library/unittest.html>`_ as testing frameworks
-4. Preconfigured `tox <https://tox.readthedocs.io/en/latest/>`_ to run pytest matrices with different Python environments
-5. Preconfigured `readthedocs <https://readthedocs.org/>`_
-6. Eleven Github workflows:
+4. `nox <https://nox.thea.codes/en/stable/>`_ to automate testing in multiple Python environments
+5. `pre-commit <https://pre-commit.com/>`_ to run various code style linters and to enforce a common style
+6. Preconfigured `readthedocs <https://readthedocs.org/>`_
+7. Eleven Github workflows:
```
Comment (Imipenem) [inline docs/available_templates/cli_python.rst:98]: still eleven?

Comment (Zethson) [reply docs/available_templates/cli_python.rst:98]: As a wise orange once said: "STOP THE COUNT"

File: cookietemple/create/templates/common_files/{{cookiecutter.commonName}}/docs/conf.py [hunk 21]
```diff
@@ -1,50 +1,46 @@
 #!/usr/bin/env python
```
Comment (Imipenem) [inline cookietemple/create/templates/common_files/{{cookiecutter.commonName}}/docs/conf.py:1]: why all the changes from `"` to `'` ;)?

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.flake8 (hunk 3 above)
Comment (Imipenem) [reply cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.flake8:4]: Ok, accepted ;)

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/dependabot.yml (hunk 4 above)
Comment (Imipenem) [reply cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/dependabot.yml:14]: I see

File: cookietemple/create/templates/common_files/{{cookiecutter.commonName}}/docs/conf.py (hunk 21 above)
Comment (Zethson) [reply cookietemple/create/templates/common_files/{{cookiecutter.commonName}}/docs/conf.py:1]: black for this template

the template uses common files

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/cookietemple.cfg (hunk 16 above)
Comment (Imipenem) [reply cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/cookietemple.cfg:18]: Yeah sorry, saw it at the end^^ ;)

File: docs/available_templates/cli_python.rst (hunk 20 above)
Comment (Imipenem) [reply docs/available_templates/cli_python.rst:98]: Lol ;D

File: cookietemple/create/templates/common_files/{{cookiecutter.commonName}}/docs/conf.py (hunk 21 above)
Comment (Imipenem) [reply cookietemple/create/templates/common_files/{{cookiecutter.commonName}}/docs/conf.py:1]: I think you explained this already ^^

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/constraints.txt (hunk 6 above)
Comment (Imipenem) [reply cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/constraints.txt:1]: Ând they dont need to be kept up to date?

Comment (Zethson) [reply cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/constraints.txt:1]: Dependabot actually picked them up (somehow).

File: cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/publish_package.yml (hunk 8 above)
Comment (Zethson) [reply cookietemple/create/templates/cli/cli_python/{{ cookiecutter.project_slug_no_hyphen }}/.github/workflows/publish_package.yml:30]: I would say no, since pip is special in that it is being used to install poetry and to install poetry we must ensure to use the latest pip version
